### PR TITLE
Fixes #37917 - Search by title for registration parameters

### PR DESCRIPTION
**Pull request by stejskalleos**
Parameters:
* Organization
* Location
* Host group
* Operating System

Inspired by https://github.com/Katello/katello-client-bootstrap

### CODE REVIEW

**Comment by adamruzicka** [comment]
This would be a nice quality of life improvement.

_Thinking out loud, take with a grain of salt:_
At the same time, (afaik) our api accepts only ids almost everywhere. Hammer can do the resolution for you, if you're doing your own scripts then you can script around a bit to resolve names to ids yourself. I'm a little bit worried that introducing title or id lookup in a single endpoint would force us to implement it everywhere, otherwise the other endpoints would appear inconsistent and incomplete in comparison. Was this planned to be a first step or the whole thing?

**Comment by stejskalleos** [comment]
> At the same time, (afaik) our api accepts only ids almost everywhere. Hammer can do the resolution for you

The main idea behind this feature is to improve users' transition from [Katello bootstrap](https://github.com/Katello/katello-client-bootstrap) to Global Registration. With the bootstrap script, you could easily use titles, but with the generated GR command, you cannot. Hammer might be the answer if you have access and permission to use it, which might only always be the case. 
For example, a Satellite admin will create one command, which they then distribute among their colleagues for registration purposes.

> I'm a little bit worried that introducing title or id lookup in a single endpoint would force us to implement it everywhere

I hope this PR won't create a precedent for it.

> Was this planned to be a first step or the whole thing?

This is the whole thingy.

I know the RegistrationController is in the API module, but it's kinda a special kid on the block; for generating registration commands via API, we still use the `_id` parameters.

File: app/controllers/concerns/foreman/controller/registration.rb [hunk 1]
```diff
@@ -144,6 +144,16 @@ def api_authorization_token
     User.current.jwt_token!(expiration: 4.hours.to_i, scope: scope)
   end
 
+  def find_object(klass, id = nil, title = nil)
```
Comment (ShimShtein) [inline app/controllers/concerns/foreman/controller/registration.rb:148]: We can use friendlyId features here: https://github.com/norman/friendly_id?tab=readme-ov-file#usage

Comment (stejskalleos) [reply app/controllers/concerns/foreman/controller/registration.rb:148]: Can you use `friendly_id` without ActiveRecord? From the description, it looks like it's only for Rail models.

Comment (ShimShtein) [reply app/controllers/concerns/foreman/controller/registration.rb:148]: Didn't try it, but then the class won't have the `authorized`, `find` and `find_by` methods. 
I don't think it's a problem in this case.

**Comment by stejskalleos** [comment]
@ShimShtein updated, now using `friendly_id`

**Review by ShimShtein — APPROVED** [review]

**Comment by stejskalleos** [comment]
Failing tests are not related, more in https://github.com/theforeman/foreman/pull/10377

**Pull request merged**